include/drivers/gpio.h

```diff
@@ -551,42 +544,38 @@ enum gpio_int_trig {
 };
 
 struct gpio_driver_api {
-	int (*config)(struct device *port, int access_op, u32_t pin, int flags);
-	int (*write)(struct device *port, int access_op, u32_t pin,
-		     u32_t value);
-	int (*read)(struct device *port, int access_op, u32_t pin,
-		    u32_t *value);
+	int (*config)(struct device *port, gpio_pin_t pin, gpio_flags_t flags);
```

Comment:
Since you still need to fix the checkpatch warning it would be better to name this function `pin_configure` not `pin_config`. It would match the new naming scheme.